src/compiler/parser.ts

```diff
@@ -719,6 +719,7 @@ namespace ts {
             // Set source file so that errors will be reported with this file name
             sourceFile = createSourceFile(fileName, ScriptTarget.ES2015, ScriptKind.JSON, /*isDeclaration*/ false);
             sourceFile.flags = contextFlags;
+            noPragmas(sourceFile);

```

Comment:
Why is this done here. You dont want to set these for tsconfig files but set them only for modules which is done is parseSourceFile

Reply:
In `types.ts` we declare `interface JsonSourceFile extends SourceFile`. So we will sometimes have a `SourceFile` object that actually comes from a `.json` file. When we try accessing properties that are non-optional on `SourceFile`, we'll crash if the json file didn't define them.

The other option would be to make the properties added by `noPragmas` optional and handle them at the use-site, if you'd prefer that.

Reply:
I think you want to set them you want to do that in  `parseSourceFile` https://github.com/Microsoft/TypeScript/blob/master/src/compiler/parser.ts#L693

Reply:
`parseSourceFile` won't be called for a json source file though. (See the branch in `createSourceFile` that calls `parseJsonText` instead of `parseSourceFile`.) But we could move the code to `createSourceFile` which they both call, would that be better?

Reply:
@sheetalkamat Thoughts?